pkg/lib/clusterconfig/clusterconfig.go

```diff
@@ -326,17 +330,53 @@ var UserValidation = &cr.StructValidation{
 	},
 }
 
-func (cc *ClusterConfig) Validate() error {
+func (cc *ClusterConfig) Validate(accessKeyID string, secretAccessKey string) error {
 	if *cc.MinInstances > *cc.MaxInstances {
 		return ErrorMinInstancesGreaterThanMax(*cc.MinInstances, *cc.MaxInstances)
 	}
 
-	if _, ok := aws.InstanceMetadatas[*cc.Region][*cc.InstanceType]; !ok {
+	if _, ok := awslib.InstanceMetadatas[*cc.Region][*cc.InstanceType]; !ok {
 		return errors.Wrap(ErrorInstanceTypeNotSupportedInRegion(*cc.InstanceType, *cc.Region), InstanceTypeKey)
 	}
 
+	if strings.HasPrefix(*cc.InstanceType, "p") {
+		sess, err := session.NewSession(&aws.Config{
```

Comment:
How do you feel about adding this as a helper in our lib/aws package? Maybe it could take in the quota name and return true/false. The advantage would be less code in this file, and also we could go back to `aws` instead of `libaws`